pkg/api/common.go

```diff
@@ -22,8 +24,18 @@ var (
 	maxTimeFormatted = maxTime.Format(time.RFC3339Nano)
 )
 
-func setHeaders(w http.ResponseWriter, res *promql.Result, warnings storage.Warnings) {
-	w.Header().Set("Access-Control-Allow-Origin", "*")
+type Config struct {
+	AllowedOrigin *regexp.Regexp
+}
+
+func corsWrapper(conf *Config, f http.HandlerFunc) http.HandlerFunc {
```

Comment:
Design question: why do we have this as a separate wrapper? Why isn't this part of `setResponseHeaders`?

Reply:
1. I wanted to re-use the prometheus httputil.SetCORS function, so we're more future proof against prometheus changing their api code
2. could write better tests where we test if the cors settings are ok once, not part of the tests for each endpoint

Reply:
Fixed 1. About 2 i'm kinda torn. Your way would be less code but not all api endpoints do/should do this and from a layering point of view perhaps the api endpoints should know this, not the routing code

Reply:
(I think you accidentally responed to Blagoj instead of me)
fair; I was inclined to think the other way, but I guess I buy that it's part of the API.